src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -141,11 +141,11 @@ class TaskReplaceActor(
         logger.warn(s"Deployment $deploymentId: New $id is terminal ($condition) on agent $agentId during app $pathId restart: " +
           s"$condition reservation: ${instance.reservation}. Waiting for the task to restart...")
         instanceTerminated(id)
-        instancesStarted -= 1
```

Comment:
Instances are automatically rescheduled so there is no need to subtract this number. Note, we do not use `instancesStarted` to determine if we are done. Ideally we would keep a set of instance ids.